sdk/src/client/node_api/indexer/query_parameters.rs

```diff
@@ -3,343 +3,295 @@
-            Self::Governor(v) => ("governor", v.to_string()),
-            Self::HasExpiration(v) => ("hasExpiration", v.to_string()),
-            Self::HasNativeTokens(v) => ("hasNativeTokens", v.to_string()),
-            Self::HasStorageDepositReturn(v) => ("hasStorageDepositReturn", v.to_string()),
-            Self::HasTimelock(v) => ("hasTimelock", v.to_string()),
-            Self::Issuer(v) => ("issuer", v.to_string()),
-            Self::MaxNativeTokenCount(v) => ("maxNativeTokenCount", v.to_string()),
-            Self::MinNativeTokenCount(v) => ("minNativeTokenCount", v.to_string()),
-            Self::PageSize(v) => ("pageSize", v.to_string()),
-            Self::Sender(v) => ("sender", v.to_string()),
-            Self::StateController(v) => ("stateController", v.to_string()),
-            Self::StorageDepositReturnAddress(v) => ("storageDepositReturnAddress", v.to_string()),
-            Self::Tag(v) => ("tag", v.to_string()),
-            Self::TimelockedAfter(v) => ("timelockedAfter", v.to_string()),
-            Self::TimelockedBefore(v) => ("timelockedBefore", v.to_string()),
-            Self::UnlockableByAddress(v) => ("unlockableByAddress", v.to_string()),
-        }
-    }
+impl_query_parameters_methods!(BasicOutputQueryParameters);
 
-    pub(crate) fn kind(&self) -> u8 {
-        match self {
-            Self::Address(_) => 0,
-            Self::AccountAddress(_) => 1,
-            Self::CreatedAfter(_) => 2,
-            Self::CreatedBefore(_) => 3,
-            Self::Cursor(_) => 4,
-            Self::ExpirationReturnAddress(_) => 5,
-            Self::ExpiresAfter(_) => 6,
-            Self::ExpiresBefore(_) => 7,
-            Self::Governor(_) => 8,
-            Self::HasExpiration(_) => 9,
-            Self::HasNativeTokens(_) => 10,
-            Self::HasStorageDepositReturn(_) => 11,
-            Self::HasTimelock(_) => 12,
-            Self::Issuer(_) => 13,
-            Self::MaxNativeTokenCount(_) => 14,
-            Self::MinNativeTokenCount(_) => 15,
-            Self::PageSize(_) => 16,
-            Self::Sender(_) => 17,
-            Self::StateController(_) => 18,
-            Self::StorageDepositReturnAddress(_) => 19,
-            Self::Tag(_) => 20,
-            Self::TimelockedAfter(_) => 21,
-            Self::TimelockedBefore(_) => 22,
-            Self::UnlockableByAddress(_) => 23,
-        }
-    }
-}
-
-impl fmt::Display for QueryParameter {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let query_tuple = self.to_query_tuple();
-        write!(f, "{}={}", query_tuple.0, query_tuple.1)
+impl BasicOutputQueryParameters {
+    /// Sets `.address(address).has_expiration(false).has_storage_deposit_return(false).has_timelock(false)` to only
+    /// get outputs that can be unlocked by the address without potential further restrictions.
+    pub fn only_address_unlock_condition(self, address: impl Into<Bech32Address>) -> Self {
```

Comment:
I feel like this should actually create a `BasicOutputQueryParameters`, it feels weird to call this on an existing one

Reply:
Changed